File: course/section_5_binary_search/code/10-sqrt-x.py
class Solution:
    def mySqrt(self, x: int) -> int:
        left, right = 0, x
        while left <= right:
            m = (left + right) // 2
            sqr = m**2
            if sqr == x:
                return m
            if sqr > x:
                right = m - 1
            else:
                left = m + 1
        return left - 1

File: course/section_5_binary_search/code/10-sqrt-x.php
<?php

class Solution {
    function mySqrt($x) {
        $left = 0;
        $right = $x;
        while ($left <= $right) {
            $m = intdiv($left + $right, 2);
            $sqr = $m *  $m;
            if ($sqr == $x) {
                return $m;
            }
            if ($sqr > $x) {
                $right = $m - 1;
            } else {
                $left = $m + 1;
            }
        }
        return $left - 1;
    }
}
